debezium-connector-postgres/src/test/java/io/debezium/connector/postgresql/RecordsStreamProducerIT.java

```diff
@@ -142,6 +143,46 @@ public void shouldReceiveChangesForInsertsCustomTypes() throws Exception {
 
     }
 
+    @Test
+    @FixFor("DBZ-1141")
+    public void shouldProcessNotNullColumnsConnectDateTypes() throws Exception {
+        testProcessNotNullColumns(TemporalPrecisionMode.CONNECT);
+    }
+
+    @Test
+    @FixFor("DBZ-1141")
+    public void shouldProcessNotNullColumnsAdaptiveDateTypes() throws Exception {
+        testProcessNotNullColumns(TemporalPrecisionMode.ADAPTIVE);
+    }
+
+    @Test
+    @FixFor("DBZ-1141")
+    public void shouldProcessNotNullColumnsAdaptiveMsDateTypes() throws Exception {
+        testProcessNotNullColumns(TemporalPrecisionMode.ADAPTIVE_TIME_MICROSECONDS);
+    }
+
+    private void testProcessNotNullColumns(TemporalPrecisionMode temporalMode) throws Exception {
+        PostgresConnectorConfig config = new PostgresConnectorConfig(TestHelper.defaultConfig()
+                .with(PostgresConnectorConfig.INCLUDE_UNKNOWN_DATATYPES, true)
+                .with(PostgresConnectorConfig.SCHEMA_BLACKLIST, "postgis")
+                .with(PostgresConnectorConfig.TIME_PRECISION_MODE, temporalMode)
+                .build());
+        setupRecordsProducer(config);
+        TestHelper.executeDDL("postgres_create_tables.ddl");
+
+        consumer = testConsumer(1);
+        recordsProducer.start(consumer, blackHole);
+
+        executeAndWait("INSERT INTO not_null_table VALUES (default, 30, '2019-02-10 11:34:58', '2019-02-10 11:35:00', '10:20:11', '10:20:12', '2019-02-01', '$20', B'101')");
+        consumer.remove();
+
+        consumer.expects(1);
+        executeAndWait("UPDATE not_null_table SET val=40");
+        final SourceRecord record = consumer.remove();
+        VerifyRecord.isValidUpdate(record, "pk", 1);
```

Comment:
@jpechane Can't we actually test the expected fallback values here? (Assuming the table uses default replica identity, in which case the fallback values should be used for the "before" row state). WDYT?